perceval/backends/mozilla_reps.py

```diff
@@ -0,0 +1,263 @@
+    """
+    version = '0.1.0'
+
+    def __init__(self, url=None, cache=None, origin=None):
+        if not url:
+            url = "https://reps.mozilla.orghttps://reps.mozilla.org "
+        origin = origin if origin else url
+
+        super().__init__(origin, cache=cache)
+        self.url = url
+        self.client = MozillaRepsClient(url)
+
+    @metadata
+    def fetch(self):
+        """Fetch events from the MozillaReps url.
+
+        The method retrieves, from a MozillaReps url, the
+        events.
+
+        :returns: a generator of events
+        """
+
+        logger.info("Looking for events at url '%s'", self.url)
+
+        self._purge_cache_queue()
+
+        nevents = 0  # number of events processed
+        nevents_total = 0  # number of events from API data
+        more_events = True # There are more events to be processed
+        events_next_uri = None # URI for the next events query
+
+        while more_events:
+            raw_events = self.client.get_events(events_next_uri)
+            self._push_cache_queue(raw_events)
+            events_data = json.loads(raw_events)
+            try:
+                logger.debug("Reps: %i/%i", nevents, nevents_total)
+                nevents_total = events_data['meta']['total_count']
+                events_next_uri = events_data['meta']['next']
+                if not events_next_uri:
+                    more_events = False
+                events = events_data['objects']
+            except:
+                cause = ("Bad JSON format for mozilla_events: %s" % (events_data))
+                raise ParseError(cause=cause)
+            for rep in events:
+                yield rep
+                nevents += 1
+            self._flush_cache_queue()
+
+        logger.info("Total number of events: %i (%i expected)", nevents, nevents_total)
+
+    @metadata
+    def fetch_from_cache(self):
+        """Fetch the builds from the cache.
+
+        :returns: a generator of events
+
+        :raises CacheError: raised when an error occurs accessing the
+            cache
+        """
+        if not self.cache:
+            raise CacheError(cause="cache instance was not provided")
+
+        cache_items = self.cache.retrieve()
+
+        for items in cache_items:
+            events = json.loads(items)['objects']
+            for event in events:
+                yield event
+
+    @staticmethod
+    def metadata_id(item):
+        """Extracts the identifier from a Build item."""
+        return MozillaReps.__metadata_id_event(item)
+
+    @staticmethod
+    def metadata_updated_on(item):
+        """Extracts the update time from a MozillaReps item.
+
+        The timestamp is extracted from 'timestamp' field.
+        This date is a UNIX timestamp but needs to be converted to
+        a float value.
+
+        :param item: item generated by the backend
+
+        :returns: a UNIX timestamp
+        """
+        return MozillaReps.__metadata_updated_on_event(item)
+
+    @staticmethod
```

Comment:
Legacy code from and older version that supports events and representatives. Fixed.